includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -1095,8 +1095,12 @@ public function log_payment_error_ajax() {
 			}
 			$order->add_order_note( $order_note );
 
+			self::remove_upe_payment_intent_cookie();
```

Comment:
This is the only place I found to handle "card declined" and "authentication failed" scenarios.

cc @FangedParakeet

Reply:
At least in my debugging, the control never went to `WC_Payment_Gateway_WCPay::update_order_status` on client-side errors (card declined or auth failed). I just re-checked to be sure. Still nothing :(